### ISSUE TITLE
[stable/mongodb] Authentication failure with mongodb 

### ISSUE THREAD

**Issue by akrausesap** [helm/charts#21078]
**Describe the bug**
When installing stable/mongodb through `helm install  --name first-mongo --set "persistence.size=2Gi" stable/mongodb --namespace personservice` as of version 7.0.2 (and greater) the command `mongo admin --host first-mongo-mongodb --authenticationDatabase admin -u root -p $MONGODB_ROOT_PASSWORD` fails to execute inside the container with the following message:

```
I have no name!@first-mongo-mongodb-6647bdd594-6csg6:/$ mongo admin --host first-mongo-mongodb --authenticationDatabase admin -u root -p $MONGODB_ROOT_PASSWORD
MongoDB shell version v4.0.11
connecting to: mongodb://first-mongo-mongodb:27017/admin?authSource=admin&gssapiServiceName=mongodb
2020-02-27T05:10:23.587+0000 E QUERY    [js] Error: Authentication failed. :
connect@src/mongo/shell/mongo.js:344:17
@(connect):2:6
exception: connect failed
I have no name!@first-mongo-mongodb-6647bdd594-6csg6:/$ exit
exit
command terminated with exit code 1
```

**Version of Helm and Kubernetes**:
Kubernetes version is: v1.14.10-gke.17
Helm Version is: v2.10.0
Istio: 1.3.5

**Which chart**:
stable/mongodb (>= 7.0.2)

**What happened**:
Can't authenticate with MONGODB_ROOT_PASSWORD

**What you expected to happen**:
Can authenticate with MONGODB_ROOT_PASSWORD

**How to reproduce it** (as minimally and precisely as possible):
```
kubectl create ns personservice
helm install  --name first-mongo --set "persistence.size=2Gi" stable/mongodb --namespace personservice
kubectl exec -n personservice <podname> /bin/bash
mongo admin --host first-mongo-mongodb --authenticationDatabase admin -u root -p $MONGODB_ROOT_PASSWORD
```

**Anything else we need to know**:
I can observe the same behaviour as described in https://github.com/istio/istio/issues/11675

**Comment by javsalgar** [comment]
Hi,

Could you confirm if you are using istio or any sidecar injection?

**Comment by akrausesap** [comment]
@javsalgar: It is definetly related. however chart version 7.0.1 works with istio. 7.0.2 does not. My hypothesis is that the script inside the container image has changed.

**Comment by akrausesap** [comment]
Hi all, 
a potential workaround for this is to disable sidecar injection and to tell istio to not use mtls for all connections to the mongo db. with automatic sidecar injection enabled on the ns this can be achived as follows:

**Comment by javsalgar** [comment]
Hi,

We are currently in version 7.8.6, so there have been several changes in the container. You could try using the container version that was available in 7.0.1 to see if that helps. However, thanks for sharing the workaround, let's see if the isitio issue gets resolved :)
